java/src/main/java/com/logicalclocks/hsfs/engine/FeatureGroupEngine.java

```diff
@@ -102,6 +113,19 @@ public void saveFeatureGroup(FeatureGroup featureGroup, Dataset<Row> dataset, Li
     featureGroup.setCorrelations(apiFG.getCorrelations());
     featureGroup.setHistograms(apiFG.getHistograms());
 
+    /* if hudi precombine key was not provided and TimeTravelFormat is HUDI, retrieve from backend and set */
+    if (featureGroup.getTimeTravelFormat() == TimeTravelFormat.HUDI & hudiPrecombineKey == null) {
+      String hudiPrecombineFeatureName = apiFG.getFeatures().stream()
+          .filter(Feature::getHudiPrecombineKey)
+          .map(Feature::getName)
+          .collect(Collectors.toList()).get(0);
+      featureGroup.getFeatures().forEach(f -> {
+        if (f.getName().equals(hudiPrecombineFeatureName)) {
+          f.setHudiPrecombineKey(true);
+        }
+      });
+    }
```

Comment:
Honestly here I'd just replace the feature list directly. It's more efficient since you already have deserialized it.